src/Atmos/Model/bc_energy.jl

```diff
@@ -44,21 +44,29 @@ function atmos_energy_boundary_state!(
     state⁺.energy.ρe =
         E_int⁺ + state⁺.ρ * gravitational_potential(atmos.orientation, aux⁻)
 end
+
 function atmos_energy_normal_boundary_flux_second_order!(
     nf,
     bc_energy::PrescribedTemperature,
     atmos,
     fluxᵀn,
     args,
 )
-    @unpack state⁻, aux⁻, diffusive⁻, t, n⁻ = args
-
-    # TODO: figure out a better way...
-    ν, D_t, _ = turbulence_tensors(atmos, state⁻, diffusive⁻, aux⁻, t)
-    d_h_tot = -D_t .* diffusive⁻.energy.∇h_tot
-    nd_h_tot = dot(n⁻, d_h_tot)
+    @unpack state⁻, aux⁻, diffusive⁻, hyperdiff⁻, t, n⁻ = args
+    tend = Flux{SecondOrder}()
+    _args = (;
+        state = state⁻,
+        aux = aux⁻,
+        t,
+        diffusive = diffusive⁻,
+        hyperdiffusive = hyperdiff⁻,
+    )
+    pargs = merge(_args, (precomputed = precompute(atmos, _args, tend),))
+    total_flux =
+        Σfluxes(Energy(), eq_tends(Energy(), atmos, tend), atmos, pargs)
+    nd_h_tot = dot(n⁻, total_flux)
```

Comment:
I think `total_flux` is a quantity per unit volume? In that case, can we call this `nd_ρh_tot`?